### PR TITLE
Refactor for scraped

### PR DESCRIPTION
**Pull request by ondenman**
This is a step towards scraping the legislature with the new term dates. (See Senate: https://github.com/everypolitician-scrapers/argentina-senate/commit/a93b937ac40714806062e172986567091a529464)

* [x] 1. scraper is on Morph.io under the "everypolitician-scrapers" group?
* [x] 2. scraper's GitHub "Website" link points at morph.io page?
* [x] 3. scraper is set to auto-run?
* [x] 4. scraper is archiving?
* [x] 5. legislature has a scraper webhook set?

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -2,15 +2,20 @@
 # specified here will be installed and made available to your morph.io scraper.
 # Find out more: https://morph.io/documentation/ruby
 
-source "https://rubygems.org"
+source 'https://rubygems.org'
 
-ruby "2.0.0"
+ruby '2.3.1'
```
Comment (chrismytton) [inline Gemfile]: I think it probably makes sense to use the latest Ruby version, which is currently [2.3.3](https://www.ruby-lang.org/en/downloads/) :)

File: lib/member.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require 'scraped'
+
+class Member < Scraped::HTML
+  field :section do
+    @section ||= MemberSection.new(response: response, noko: noko)
```
Comment (chrismytton) [inline lib/member.rb]: I think the `@section ||=` bit here (and the `@page` version below) are probably unnecessary optimizations at this stage. Probably best to drop them for now until they prove they are needed.

Comment (chrismytton) [inline lib/member.rb]: Passing `noko: noko` is redundant as the `MemberSection` will automatically create an identical `noko` object from the response you're passing in. So you can drop that parameter.

File: lib/member.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+require 'scraped'
+
+class Member < Scraped::HTML
```
Comment (chrismytton) [inline lib/member.rb]: Hmm, I don't think it makes sense for `Member` to be a subclass of `Scraped::HTML`. You're quite right to create a `Member` class here - top level concepts like `Member` should be reserved for classes that pull together different concepts.

I think for this we probably want a plain `Member` class without a parent which implements a `#to_h` method which returns the information from the `MemberSection` class with the information from the `MemberPage` class.

File: lib/member_page.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+require 'scraped'
+
+class MemberPage < Scraped::HTML
+  field :name do
+    noko.xpath('//div[@class="col-sm-12 col-md-2 distrito"]/following-sibling::div[@class="col-sm-12 col-md-4"]/h1/text()')
```
Comment (chrismytton) [inline lib/member_page.rb]: This and the selectors below seem to be very specific. Are the bootstrap classes (`col-*`) actually _required_ for this to work?

File: lib/member_section.rb [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  field :party do
+    noko.xpath('following::td')[3].text.tidy
+  end
+
+  field :term do
+    133
+  end
+
+  field :source do
+     URI.join(url, noko.attr('href')).to_s
```
Comment (chrismytton) [inline lib/member_section.rb]: Looks like the indentation has gone a bit awry here :)

File: scraper.rb [hunk 6]
```diff
@@ -7,55 +7,25 @@
...
-    phone: noko.css('div.info-diputados-principal1').text[/Teléfono: (.*)$/, 1].to_s.tidy,
-    email: noko.css('div.info-diputados-principal2 a[href*="/contacto"]').text.tidy,
-  }
-  return data
+url = 'http://www.hcdn.gob.ar/diputados/listadip.html'
+MembersPage.new(response: Scraped::Request.new(url: url).response).members.each do |member|
+  data = [
+    member.section,
+    member.page
+  ].map(&:to_h).reduce(&:merge)
```
Comment (chrismytton) [inline scraper.rb]: I think this logic can be pushed down into the `Member` class as discussed in my comments above.

**Review by chrismytton — CHANGES_REQUESTED** [review]

**Comment by ondenman** [comment]
Closing this as it's a bit long winded. I've opened an issue in ep-data https://github.com/everypolitician/everypolitician/issues/601

**Pull request closed without merging**